### DISCUSSION TITLE
Installing Dependencies

### DISCUSSION THREAD

**Discussion by BillyDodds** [EY-Data-Science-Program/2022-Better-Working-World-Data-Challenge#5] [General]
Hi froggers, 

There have been a few people having issues installing the dependencies required for both challenges. If this is the case, try using `conda env create -n 2022DSC --file <folder_path>/notebooks/environment.yml` on the new environment.yml file recently uploaded to the git repo. 

If using the Azure VM deployment, installing using the requirements.txt file via `conda create -n 2022DSC --file <folder_path>/notebooks/requirements.txt` should hopefully work. Otherwise, please try the .yml approach above. 

If any issues with package installations arise, my suggestion would be to examine the requirements.txt file in the repo, identify the version of the package causing conflicts, and install that specific version via `conda install <package_name>=<package_version>`. 

Please direct any package installation issues to this thread and we will try our best to assist. 

All the best! 
Billy Dodds

**Comment by ai-perture** [comment]
Hello, 
I have a package issue with folium in Azure. It has been installed (i think, i'm not sure how to check), but when running "import folium" in notebooks, i get the error "No module named 'folium' ". How can I solde this? 
Thanks, 

Ai-Ling Bonnet

**Reply by BillyDodds** [reply]
Hi @ai-perture, 

The folium package is not in the requirements.txt file because it is only relevant for the supplementary notebooks. Its likely that it is not installed. Try opening a terminal in JupyterHub and entering the following commands: 
```sh
conda activate 2022DSC
```
(or whatever you set your virtual environment's name to be), then:

```sh
conda install folium
```

Then restart your notebook's kernel and rerun the notebook. Hopefully, that works. 
 
Cheers, 
Billy Dodds

**Comment by jonasanmaru** [comment]
I am still facing the same issue, could it be related to the anaconda version or python version? 
*I am using the latest anaconda from the website provided in the wiki guide. 
![image](https://user-images.githubusercontent.com/16115274/168738092-40d15537-f78f-48d2-87ae-011db8d670aa.png)

**Reply by BillyDodds** [reply]
Hi @jonasanmaru, 

Was this after you reinstalled the packages with the environment.yml or the requirements.txt file in the way I outlined in this discussion post?

Cheers, 
Billy

**Reply by jonasanmaru** [reply]
Hi @BillyDodds ,

Yes, correct, I clone the entire repository again, and ran the installation via envionment.yml.
`conda env create -n 2022DSC --file ./notebooks/environment.yml`

[error.txt](https://github.com/EY-Data-Science-Program/2022-Better-Working-World-Data-Challenge/files/8705862/error.txt)

**Reply by BillyDodds** [reply]
This is quite strange. It looks like nearly all the packages are conflicting for you, but when I install it on my local environment I get no conflicts. 

The only thing I can think of is checking the conda version. On the Azure VM, we use conda 4.12.0. And on my local Mac I use 4.10.3. Maybe try a different version of conda?

**Reply by jonasanmaru** [reply]
Hi @BillyDodds ,

My conda information is as follows:
conda version : 4.12.0
conda-build version : 3.21.8
python version : 3.9.12.final.0

I also tried downgrading from 4.12 to 4.10.3, same issues arised.
I think this was related to some bug as mentioned in stackoverflow
https://stackoverflow.com/questions/58216917/create-conda-environment-found-conflicts-when-solving-environment-and-findi

I will try with WSL2 and see if it works. 
Thank you for all help.

**Comment by sendilc** [comment]
Hi, 

I am getting the following error while creating a new virtual environment ... I have tried the absolute and relative path (onda env create -n 2022DSC --file <folder_path>/notebooks/environment.yml) but still facing this issue.  

Collecting package metadata (repodata.json): failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.anaconda.com/pkgs/main/win-64/repodata.json>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

If your current network has https://www.anaconda.com blocked, please file
a support request with your network engineering team.

'https://repo.anaconda.com/pkgs/main/win-64'

**Reply by BillyDodds** [reply]
Hi @sendilc. Looks like the issue is to do with your network firewall. Can you access https://www.anaconda.com/ in your browser?

**Reply by sendilc** [reply]
Hi @BillyDodds , Yes I was able to access https://www.anaconda.com/ in browser? I did a bit of google and eventually was able to setup the environment but updating the conda version. Thanks for your help and support.

**Comment by rahul0906** [comment]
Hi, 
Actually when I had setup whole local setup on conda, it went well without any errors. But I am not able to see jupyter Notebook Application in the drop down list of all the applications installed as in other environments. Can u please help?
